### PR TITLE
Refactor access controllers to their own module

### PR DESCRIPTION
**Pull request by haadcode**
This PR will make OrbitDB use orbit-db-access-controllers module and move the code worked in various PRs orbitdb/orbit-db to their own repo. This PR is to the feat/new-ipfs-log branch to continue the PR in https://github.com/orbitdb/orbit-db/pull/449.

### CODE REVIEW

**Comment by haadcode** [comment]
This PR also takes in changes in https://github.com/orbitdb/orbit-db/pull/490.

Will resolve the merge conflicts.

File: package.json [hunk 1]
```diff
@@ -16,28 +16,30 @@
...
     "orbit-db-feedstore": "~1.4.0",
-    "orbit-db-identity-provider": "github:orbitdb/orbit-db-identity-provider",
+    "orbit-db-identity-provider": "github:orbitdb/orbit-db-identity-provider#feat/verify-identity",
     "orbit-db-keystore": "github:orbitdb/orbit-db-keystore#fix/verification",
     "orbit-db-kvstore": "~1.4.0",
     "orbit-db-pubsub": "~0.5.5",
     "orbit-db-store": "github:shamb0t/orbit-db-store#feat/new-ipfs-log"
   },
   "devDependencies": {
+    "@colony/purser-software": "^1.1.2",
```
Comment (haadcode) [inline package.json]: Not needed I believe, moved to https://github.com/orbitdb/orbit-db-access-controllers

File: package.json [hunk 2]
```diff
@@ -48,6 +50,7 @@
     "remark-validate-links": "^7.0.0",
     "rimraf": "^2.6.2",
     "uglifyjs-webpack-plugin": "^1.1.4",
+    "web3": "^1.0.0-beta.36",
```
Comment (haadcode) [inline package.json]: Not needed, moved to https://github.com/orbitdb/orbit-db-access-controllers

File: src/OrbitDB.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 'use strict'
 
 const path = require('path')
+const pMap = require('p-map')
```
Comment (haadcode) [inline src/OrbitDB.js]: Is this needed?

File: src/OrbitDB.js [hunk 4]
```diff
@@ -45,6 +46,8 @@ let databaseTypes = {
     this.directory = options.directory || './orbitdb'
     this.stores = {}
     this._directConnections = {}
+
+    ACFactory = options.ACFactory || ACFactory
```
Comment (haadcode) [inline src/OrbitDB.js:49]: Note, this is really needed only for testing purposes, OrbitDB should always use orbit-db-access-controllers as the ACFactory

File: src/OrbitDB.js (hunk 3 above)
Comment (shamb0t) [reply src/OrbitDB.js]: Not anymore, was needed for adding keys to smart-contract but can be removed now

File: src/utils/ensure-ac-address.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+'use strict'
```
Comment (haadcode) [inline src/utils/ensure-ac-address.js]: his is not needed anymore, moved to https://github.com/orbitdb/orbit-db-access-controllers

File: test/custom-keystore.test.js [hunk 6]
```diff
@@ -59,7 +59,7 @@ const databases = [
 ]
 
 Object.keys(testAPIs).forEach(API => {
-  describe(`orbit-db - Use a Custom Keystore (${API})`, function() {
+  describe.skip(`orbit-db - Use a Custom Keystore (${API})`, function() {
```
Comment (haadcode) [inline test/custom-keystore.test.js:62]: Note, skip for now, fix in main PR

File: test/eventlog.test.js [hunk 7]
```diff
@@ -87,7 +87,7 @@ Object.keys(testAPIs).forEach(API => {
 
       it('adds an item that is > 256 bytes', async () => {
         db = await orbitdb1.eventlog('third database')
-        let msg = new Buffer(1024)
+        let msg = Buffer.alloc(1024)
```
Comment (haadcode) [inline test/eventlog.test.js:90]: New Buffer is deprecated starting node.js 10, switch to new Buffer api

File: test/utils/custom-test-keystore.js [hunk 8]
```diff
@@ -44,6 +44,7 @@ class CustomTestKeystore {
   }
 
   verify(signature, key, data) {
+    console.log(signature, key, data)
```
Comment (haadcode) [inline test/utils/custom-test-keystore.js]: remove

File: src/OrbitDB.js [hunk 9]
```diff
@@ -34,7 +34,7 @@ let databaseTypes = {
       throw new Error('IPFS is a required argument. See https://github.com/orbitdb/orbit-db/blob/master/API.md#createinstance')
 
     if (!isDefined(identity))
-      throw new Error('identity is a required as argument. See https://github.com/orbitdb/orbit-db/blob/master/API.md#createinstance')
+      throw new Error('identity is a required argument. See https://github.com/orbitdb/orbit-db/blob/master/API.md#createinstance')
```
Comment (shamb0t) [inline src/OrbitDB.js:37]: :+1:

**Pull request merged**